### PR TITLE
add missing unique key to fragment created with map

### PR DESCRIPTION
**Pull request by zeotuan**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Not really a Bug but while I found out that some of the Preach Element that are created using map function was missing unqiue key which is not required but will allow preach to determine DOM change so it can efficiently update the DOM
## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/articles/components/CommentsList.jsx [hunk 1]
```diff
@@ -36,7 +36,12 @@ export const CommentsList = ({ comments = [], articlePath, totalCount }) => {
     return (
       <div className="crayons-story__comments">
         {comments.slice(0, numberOfCommentsToShow).map((comment) => {
-          return <CommentListItem comment={comment} />;
+          return (
+            <CommentListItem
+              key={`${comment.published_at_int}-${comment.name}`}
```
Comment (nickytonline) [inline app/javascript/articles/components/CommentsList.jsx:41]: Refactor (blocking): Use the comment ID property available in the comment object. It is guaranteed to be unique.

File: app/javascript/chat/ChatChannelSettings/MembershipManager/MembershipSection.jsx [hunk 2]
```diff
@@ -32,17 +34,7 @@ export const MembershipSection = ({
...
-      name: PropTypes.string.isRequired,
-      membership_id: PropTypes.number.isRequired,
-      user_id: PropTypes.number.isRequired,
-      role: PropTypes.string.isRequired,
-      image: PropTypes.string.isRequired,
-      username: PropTypes.string.isRequired,
-      status: PropTypes.string.isRequired,
-    }),
-  ).isRequired,
+  memberships: PropTypes.arrayOf(defaultMembershipPropType).isRequired,
```
Comment (nickytonline) [inline app/javascript/chat/ChatChannelSettings/MembershipManager/MembershipSection.jsx:37]: Nitpick (blocking): Move this change to a separate pull request as it's not related to adding missing keys in mapped components.

File: app/javascript/chat/RequestManager/ChannelRequestSection.jsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ export const ChannelRequestSection = ({
         channelRequests.map((channelPendingRequest) => {
           return (
             <RequestListItem
+              key={`${channelPendingRequest.channel_name}-${channelPendingRequest.membership_id}`}
```
Comment (nickytonline) [inline app/javascript/chat/RequestManager/ChannelRequestSection.jsx:23]: Refactor (blocking): `channel_name` is adequate for the key, as the context of this rendered component is for a specific user.

File: app/javascript/chat/RequestManager/PersonalInvitationSection.jsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ export const PersonalInvitationSection = ({
         userInvitations.map((userInvitation) => {
           return (
             <PendingInvitationListItem
+              key={`${userInvitation.channel_name}-${userInvitation.membership_id}`}
```
Comment (nickytonline) [inline app/javascript/chat/RequestManager/PersonalInvitationSection.jsx:22]: Refactor (blocking): `channel_name` is adequate for the key, as the context of this rendered component is for a specific user.

**Review by nickytonline — DISMISSED** [review]
This is looking great @zeotuan. Some comments and request changes.

**Comment by citizen428** [comment]
Note that this PR contains changes to chat components which we should undo because chat is about to be removed and we don't want to create merge conflicts. @zeotuan Are you planning on finishing this PR? :-)

**Comment by aitchiss** [comment]
Hey @zeotuan 👋  just checking in to see if you are still working on this one?

**Comment by citizen428** [comment]
I merged `main` back into this branch. After cleaning up the chat components and resolving merge conflicts there are now only changes to two remaining files.

**Review by citizen428 — APPROVED** [review]

**Review by aitchiss — APPROVED** [review]
